src/controllers/user/Controller.ts

```diff
@@ -38,44 +39,45 @@ class UserController {
       }
     }
 
-    create(req: Request, res: Response, next: NextFunction) {
-      try {
-        console.log('Inside post method of User Controller');
-        res.send({
-          message: 'User created successfully',
-          data: [
-            {
-              name: 'User1',
-            },
-            {
-                name: 'User2',
-            }
-          ]
-        });
-      } catch (err) {
-        console.log('Inside err', err);
-      }
-    }
+    public async create(req: IRequest, res: Response, next: NextFunction) {
+      const { id, email, name, role, password } = req.body;
+      const creator = req.userData._id;
 
-    update(req: Request, res: Response, next: NextFunction) {
-      try {
-        console.log('Inside update method of User Controller');
+      const user = new UserRepository();
+      await user.createUser({ id, email, name, role, password }, creator)
+          .then(() => {
+              res.send({
```

Comment:
Don't use both async-await and then() only use one to handle async operation either then() or async-await.